### DISCUSSION TITLE
'declare' won't work

### DISCUSSION THREAD

**Discussion by taspargo** [hneemann/Digital#1204] [Q&A]
I'm trying to use 'declare' to split off a bit from a bus, just like in the Help. It's giving me an odd error which isn't any help:

Error running the tests:
During the execution of the tests "Test.error" an error has occurred!
caused by: During the execution of the tests "Test.error" an error has occurred!

This recursive error almost put me in a mental hospital :)

 I made a simplified test file that reproduces this error:
```
EClk.P2 EAB7 ~YLO
declare EAB7 = (EAB>>7)&1;
C	1	0
```

I reproduced the error in a completely different circuit as well.

Attached are screenshots of the test text and the error message.

Version info:
OSX, 10.16
JRE: 13.0.1
VM: 13.0.1+9

Digital.jar
Build Git-Revision: v0.30-18-g1204c26
Build Time: 2023-09-19 05:12

<img width="361" alt="Test text" src="https://github.com/hneemann/Digital/assets/46286768/3e8c0e44-618d-4d3b-b054-60e9dd6d61fa">
<img width="550" alt="Test error" src="https://github.com/hneemann/Digital/assets/46286768/4f008dbd-79cd-4d48-bf71-f6fd5bb50905">

**Comment by taspargo** [comment]
Here's a full test case.
[Test_case_declare.dig.zip](https://github.com/hneemann/Digital/files/12778352/Test_case_declare.dig.zip)

**Comment by hneemann** [comment]
There is a bug in passing the complete error message to the Error dialog. 
Your problem is that declare cannot access the model inputs. Only the outputs can be accessed.
Inputs can always be listed directly in the header of the test. Outputs would have to be created in the circuit first, which would be inconvenient.

**Comment by taspargo** [comment]
I'm not following- if you have an input bus, how can inputs be listed in
the test header? Or do you to have inputs come in to that circuit
separately instead of as a bus?
